pkg/ingester/instance_limits.go

```diff
@@ -46,12 +47,38 @@ func (l *InstanceLimits) RegisterFlags(f *flag.FlagSet) {
 }
 
 // Sets default limit values for unmarshalling.
-var defaultInstanceLimits *InstanceLimits
+var (
+	defaultInstanceLimits    *InstanceLimits
+	defaultInstanceLimitsMtx sync.RWMutex
+)
+
+func setDefaultInstanceLimits(l *InstanceLimits) {
+	defaultInstanceLimitsMtx.Lock()
+	defer defaultInstanceLimitsMtx.Unlock()
+	defaultInstanceLimits = l
+}
+
+func getDefaultInstanceLimits() *InstanceLimits {
+	defaultInstanceLimitsMtx.RLock()
+	defer defaultInstanceLimitsMtx.RUnlock()
+
+	if defaultInstanceLimits == nil {
+		return nil
+	}
+
+	return &InstanceLimits{
+		MaxIngestionRate:        defaultInstanceLimits.MaxIngestionRate,
+		MaxInMemoryTenants:      defaultInstanceLimits.MaxInMemoryTenants,
+		MaxInMemorySeries:       defaultInstanceLimits.MaxInMemorySeries,
+		MaxInflightPushRequests: defaultInstanceLimits.MaxInflightPushRequests,
+	}
+}
```

Comment:
No need to make this complicated. Previous code was fine here, please revert these changes (drop mutex, no need for `getDefaultInstanceLimits`)

Reply:
It's effectively immutable after setting.

Reply:
> It's not complicated

We're effective doing `var il InstanceLimits; il = *defaultInstanceLimits` here. Just with 15 lines of code.